ibis/expr/datatypes.py

```diff
@@ -366,6 +372,42 @@ def can_implicit_cast(cls, other):
         return isinstance(other, (Floating, Decimal))
 
 
+_VALID_TIMEZONES = frozenset(pytz.all_timezones)
+
+
+@parametric
+class TimestampWithTimezone(DataType):
```

Comment:
I didn't explore this too much, but I could look into doing it that way. One of the reasons I did it that way was to avoid having to deal with a type that was sometimes parametric and sometimes not.